runhouse/resources/hardware/on_demand_cluster.py

```diff
@@ -464,6 +467,38 @@ def teardown(self):
         Example:
             >>> rh.ondemand_cluster("rh-cpu").teardown()
         """
+        # TODO [SB]: remove the den_auth check once we will get status of clusters without den_auth as well.
+        if self.den_auth:
+            try:
+                # cluster_config = self.config()
+                cluster_status_data = self.status()
+                status_data = {
+                    "status": "terminated",
+                    "resource_type": cluster_status_data.get("cluster_config").get(
+                        "resource_type"
+                    ),
+                    "data": cluster_status_data,
+                }
+                cluster_uri = rns_client.format_rns_address(self.rns_address)
+                api_server_url = cluster_status_data.get("cluster_config").get(
+                    "api_server_url", rns_client.api_server_url
+                )
+                post_status_data_resp = requests.post(
+                    f"{api_server_url}/resource/{cluster_uri}/cluster/status",
+                    data=json.dumps(status_data),
+                    headers=rns_client.request_headers(),
+                )
+                assert post_status_data_resp.status_code == 200
+            except Exception:
+                warning_msg = "Failed to update in Den that the cluster is terminated."
+                if post_status_data_resp:
```

Comment:
for this try / except  - the handling of an exception in the logic here can be separated from  response with a non 200 status code, but right now they look bundled together